dev-docs/RFCs/v1.0/generic-edit-mode.md

```diff
@@ -0,0 +1,93 @@
+# Generic EditMode
+
+* **Author**: Clay Anderson
+
+## Summary
+
+Create a generic `EditMode` class in `@nebula/core` that is independent of deck.gl, react-map-gl, and GeoJSON. This generic class can then be integrated into `EditableGeoJsonLayer` as well as the [upcoming DrawControl feature of react-map-gl](https://github.com/uber/react-map-gl/issues/734)
```

Comment:
`@nebula.gl/core` or `@nebula.gl/mode-handlers`

Reply:
Wouldn't it be nicer to just call these `EditingModes`
Then if you export them to the user, they can just import and add `EditingModes` to their layers.

It seems to me that `Mode-Handlers` may make sense to developers but not to users...

Reply:
I agree. I want to do this in a few steps:

1. Refactor the mode handlers to extend `EditMode`
2. Rename the mode handlers.
3. Consolidate `EditableGeoJsonLayer.mode` and `EditableGeoJsonLayer.modeHandlers` into just `EditableGeoJsonLayer.mode` and have it take an`EditMode` implementation. So there'd be no string to mode handlers mapping.

I'll add these details to the rfc